MapboxDirections/MBRouteOptions.swift

```diff
@@ -325,10 +325,10 @@ open class RouteOptions: NSObject, NSSecureCoding {
      - returns: A tuple containing an array of waypoints and an array of routes.
      */
     internal func response(_ json: JSONDictionary) -> ([Waypoint]?, [Route]?) {
-        let waypoints = (json["waypoints"] as? [JSONDictionary])?.map { waypoint -> Waypoint in
+        let waypoints = (json["waypoints"] as? [JSONDictionary])?.enumerated().map { (index, waypoint) -> Waypoint in
             let location = waypoint["location"] as! [Double]
             let coordinate = CLLocationCoordinate2D(geoJSON: location)
-            return Waypoint(coordinate: coordinate, name: waypoint["name"] as? String)
+            return Waypoint(coordinate: coordinate, name: self.waypoints[index].name ?? waypoint["name"] as? String)
```

Comment:
Since `index` marches forward predictably, we can avoid repeatedly indexing into the `waypoints` array by zipping `waypoints` and `json["waypoints"]` together.